### PR TITLE
UI Fix to give the resolution message

### PR DESCRIPTION
**Pull request by vviveksharma**
### :nut_and_bolt: Description: What code changed, and why?
- The UI was not giving the resolution message that were thrown from trigger or Handler funtions and there resolution in the how to resolve message
- the Hardware tigger and NFS mount where using same util it was giving the error response as string ( "400: Bad Request ) otherwise it should be the error that is coming from the Handlers , also refactored both the trigger files as well 

### :chains: Related Resources
- https://chefio.atlassian.net/browse/CHEF-3796
### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable

- The response from the trigger is now shown in the How Resolve section 

Screenshot 2023-06-27 at 7 46 59 PM

- The Error given from the Handlers are also captured now in the error section

Screenshot 2023-06-27 at 7 47 38 PM

Screenshot 2023-06-27 at 7 47 41 PM

Screenshot 2023-06-27 at 7 48 26 PM

### CODE REVIEW

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/nfsmountbackupchecktrigger/nfsmountbackupcheck.go [hunk 1]
```diff
@@ -76,20 +80,23 @@ func (ss *NfsBackupConfigCheck) triggerCheckForMountService(body models.NFSMount
...
 	}
 	return apiResp, nil
 }
 
 // constructErrorResult constructs the error response when recived from the API
-func constructErrorResult(config *models.Config, err error) []models.CheckTriggerResponse {
+func constructErrorResult(config *models.Config, errMessage string, statusCode int) []models.CheckTriggerResponse {
 	var result []models.CheckTriggerResponse
-
+	fmt.Errorf("Inside the custruct error=%v", errMessage)
```
Comment (vivekshankar1) [inline components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/nfsmountbackupchecktrigger/nfsmountbackupcheck.go:95]: Please remove this log

File: components/automate-cli/pkg/verifyserver/utils/checkutils/checkutil_test.go [hunk 2]
```diff
@@ -44,9 +44,10 @@ func TestPrepareTriggerResponse_WithNoError(t *testing.T) {
 	check := "test check"
 	msg := "test message"
 	isError := false
+	statusCode := 200
 
 	// Act
-	result := PrepareTriggerResponse(resp, host, nodeType, errorString, check, msg, isError)
+	result := PrepareTriggerResponse(resp, host, nodeType, errorString, check, msg, isError, statusCode)
```
Comment (vivekshankar1) [inline components/automate-cli/pkg/verifyserver/utils/checkutils/checkutil_test.go:50]:
```suggestion
	result := PrepareTriggerResponse(resp, host, nodeType, errorString, check, msg, isError, fiber.StatusOK)
```
you can directly use fiber.StatusOK

File: components/automate-cli/pkg/verifyserver/utils/checkutils/checkutil_test.go [hunk 3]
```diff
@@ -68,9 +69,9 @@ func TestPrepareTriggerResponse_WithError(t *testing.T) {
 	check := "test check"
 	msg := "test message"
 	isError := true
-
+	statusCode := 400
 	// Act
-	result := PrepareTriggerResponse(resp, host, nodeType, errorString, check, msg, isError)
+	result := PrepareTriggerResponse(resp, host, nodeType, errorString, check, msg, isError, statusCode)
```
Comment (vivekshankar1) [inline components/automate-cli/pkg/verifyserver/utils/checkutils/checkutil_test.go:74]:
```suggestion
	result := PrepareTriggerResponse(resp, host, nodeType, errorString, check, msg, isError, fiber.StatusBadRequest)
```
you can directly use like this

File: components/automate-cli/cmd/chef-automate/verify.go [hunk 4]
```diff
@@ -606,6 +606,18 @@ func buildReports(batchCheckResults []models.BatchCheckResult) []reporting.Verif
 	for _, batchCheckResult := range batchCheckResults {
 
 		for _, test := range batchCheckResult.Tests {
+
+			var errorMsgs, resolutionMsgs []string
+			var successfulCount, failedCount int
+
+			if test.Skipped {
```
Comment (prasad927) [inline components/automate-cli/cmd/chef-automate/verify.go:613]: I think line 613 will cause problem to print skipped checks on UI

File: components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/nfsmountbackupchecktrigger/nfsmountbackupcheck.go (hunk 1 above)
Comment (vviveksharma) [reply components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/nfsmountbackupchecktrigger/nfsmountbackupcheck.go]: Done removed the fat statement

File: components/automate-cli/cmd/chef-automate/verify.go (hunk 4 above)
Comment (vviveksharma) [reply components/automate-cli/cmd/chef-automate/verify.go]: done

**Review by vivek-yadav — APPROVED** [review]

**Pull request merged**